crates/ironrdp-web/src/session.rs

```diff
@@ -522,17 +534,38 @@ async fn connect(
     proxy_auth_token: String,
     destination: String,
     pcb: Option<String>,
+    kdc_proxy_url: Option<String>,
 ) -> Result<(connector::ConnectionResult, WebSocketCompat), IronRdpError> {
     let mut framed = ironrdp_futures::SingleThreadedFuturesFramed::new(ws);
 
-    let mut connector = connector::ClientConnector::new(config)
-        .with_server_name(&destination)
-        .with_credssp_network_client(WasmNetworkClientFactory);
-    // .with_static_channel(ironrdp::dvc::Drdynvc::new()); // FIXME: drdynvc is not working
-
-    let upgraded = connect_rdcleanpath(&mut framed, &mut connector, destination, proxy_auth_token, pcb).await?;
-
-    let connection_result = ironrdp_futures::connect_finalize(upgraded, &mut framed, connector).await?;
+    let mut connector = connector::ClientConnector::new(config);
+
+    let (upgraded, server_public_key) =
+        connect_rdcleanpath(&mut framed, &mut connector, destination.clone(), proxy_auth_token, pcb).await?;
+
+    info!("kdc url = {:?}", &kdc_proxy_url);
+
+    let mut wasm_network_client = WasmNetworkClient::new();
+    let network_client: Option<&mut dyn AsyncNetworkClient> = kdc_proxy_url
+        .as_ref()
+        .map(|_| &mut wasm_network_client as &mut dyn AsyncNetworkClient);
+
+    let connection_result = ironrdp_futures::connect_finalize(
+        upgraded,
+        &mut framed,
+        (&destination).into(),
+        server_public_key,
+        network_client,
+        connector,
+        Some(KerberosConfig {
+            kdc_proxy_url: kdc_proxy_url
+                .map(|url| url::Url::parse(&url))
+                .transpose()
+                .context("invalid KDC URL")?,
+            hostname: Some(destination),
```

Comment:
IIRC, this is not supposed to be the "destination" computer name, but the client computer name. There is no way to easily retrieve the computer name from the browser, and if this is a workaround, you should explicitly mark it as such:
```suggestion
            // HACK: It’s supposed to be the computer name of the client, but since it’s not easy to retrieve the information, we set the destination hostname
            // because it happens to work.
            hostname: Some(destination),
```